extensions/sqldatabase/driver/commonSql.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2022 EMQ Technologies Co., Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package driver
+
+import (
+	"fmt"
+	"github.com/lf-edge/ekuiper/pkg/cast"
+)
+
+type QueryGenerator struct {
+	indexSlice []interface{}
+	*InternalSqlQueryCfg
+}
+
+func (q *QueryGenerator) quoteIdentifier(identifier string) string {
+	return "`" + identifier + "`"
+}
+
+func (q *QueryGenerator) getSelect() string {
+	return "select * from " + q.Table + " "
+}
+
+func (q *QueryGenerator) getCondition() string {
+	var val string
+	if q.IndexField != "" {
+		if q.IndexFieldType == DATETIME_TYPE && q.DateTimeFormat != "" {
+			t, _ := cast.InterfaceToTime(q.IndexValue, q.DateTimeFormat)
+			val, _ = cast.FormatTime(t, q.DateTimeFormat)
+		} else {
+			val = fmt.Sprintf("%v", q.IndexValue)
+		}
+		return "where " + q.IndexField + " > " + q.quoteIdentifier(val) + " "
+	}
+
+	return ""
+}
+
+func (q *QueryGenerator) getOrderby() string {
+	if q.IndexField != "" {
+		return "order by " + q.quoteIdentifier(q.IndexField) + " ASC "
+	}
+	return ""
+}
+
+func (q *QueryGenerator) getLimit() string {
+	if q.Limit != 0 {
+		return fmt.Sprintf("limit %d", q.Limit)
+	}
+	return ""
+}
+
+func NewInternalSqlQuery(cfg *InternalSqlQueryCfg) SqlQueryGenerator {
+	in := &QueryGenerator{
+		InternalSqlQueryCfg: cfg,
+	}
+	return in
+}
+
+func (q *QueryGenerator) SqlQueryStatement() (string, error) {
+	return q.getSelect() + q.getCondition() + q.getOrderby() + q.getLimit(), nil
+}
+
+func (q *QueryGenerator) CollectIndexValue(row map[string]interface{}) {
+	if q.indexSlice == nil {
```

Comment:
Is this needed? Why not just get the row's last value

Reply:
ack